### ISSUE TITLE
Need to expose hot-restart and hot-reload hooks for flutter plugins

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#7160]
Plugins will want to do something at least for hot-restart.  @johnmccutchan

**Comment by mit-mit** [comment]
@mravn-google do we know if any hooks we need?

**Comment by johnmccutchan** [comment]
As a start you probably want to be notified of the following:

*) A hot reload occurred. There probably isn't anything that a native plugin has to do here but it may take this as an opportunity to clear caches / reset state.

*) A hot restart occurred. At this point the original Isolate has been killed and a new one spawned in it's place. At this point a native plugin might want to run initialization logic again.

**Comment by a-siva** [comment]
@devoncarew I presume this is already taken care of right?

**Comment by devoncarew** [comment]
I don't believe theres's any notifications for plugins; @mravn-google or @mit-mit may know for sure.

**Comment by mravn-google** [comment]
There are none, to my knowledge. And we do not seem to need any. For a plugin to support re-spawn of the Dart isolate, it just has to be written so that 1) it is the Dart code that initiates Dart/platform communication (typically via an initializing invocation across a MethodChannel), and 2) the platform side accepts more than one such initializing call. Reinitialization of the platform side after hot restart then happens on receipt of each initializing call after the first one.

**Comment by Yongle-Fu** [comment]
need expose api to close  ble gatt connection
